src/server/api_service.rs

```diff
@@ -1,43 +1,41 @@
-use cfg_if::cfg_if;
+use crate::host::{get_host, get_https};
+use actix_proxy::{IntoHttpResponse, SendRequestError as ProxyError};
+use actix_web::{
+  http::uri::{InvalidUri, InvalidUriParts, Uri},
+  web,
+  HttpRequest,
+  HttpResponse,
+};
+use awc::{
+  error::{HttpError, InvalidUrl, SendRequestError},
+  Client,
+};
 
-cfg_if! {
-    if #[cfg(feature = "ssr")] {
-        use actix_web::{HttpRequest, HttpResponse, web, http::uri::{Uri, InvalidUri, InvalidUriParts}};
-        use actix_proxy::{IntoHttpResponse, SendRequestError as ProxyError};
-        use awc::{Client, error::{InvalidUrl, SendRequestError, HttpError}};
-        use crate::host::get_host;
+pub async fn route_to_api(
+  request: HttpRequest,
+  client: web::Data<Client>,
+  payload: web::Payload,
+  uri: Uri,
+) -> Result<HttpResponse, ProxyError> {
+  let mut parts = uri.into_parts();
+  parts.authority = Some(get_host().try_into().map_err(map_uri_err::<InvalidUri>)?);
+  parts.scheme = Some(
+    format!("http{}", if get_https() == "true" { "s" } else { "" })
```

Comment:
GJ making these external requests automatically use the browser url. Much easier than having to define another env.

Reply:
oh i think that logic was already in there :smile: .. but it reminds me of the case when you're using the proxy it should adopt the protocol the page is using, another tiny PR incoming!

would be awesome if PaginationCursor can be easily to/from string, then i can page the list of posts from the server side :+1: